Public/Src/Tools/Tool.MsBuildGraphBuilder/MsBuildPredictionCollector.cs

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Build.Prediction;
+
+namespace MsBuildGraphBuilderTool
+{
+    /// <summary>
+    /// Collects predictions from the MsBuild prediction library.
+    /// </summary>
+    /// <remarks>
+    /// This validates the predicted inputs and outputs, in case the predictor is not working properly
+    /// </remarks>
+    public sealed class MsBuildPredictionCollector : IProjectPredictionCollector
+    {
+        private readonly ConcurrentQueue<(string predictorName, string failure)> m_predictionFailures;
+
+        private readonly ICollection<string> m_inputFilePredictions;
+
+        private readonly ICollection<string> m_outputFolderPredictions;
+
+        public MsBuildPredictionCollector(
+            ICollection<string> inputFilePredictions,
+            ICollection<string> outputFolderPredictions,
+            ConcurrentQueue<(string predictorName, string failure)> predictionFailures)
+        {
+            m_inputFilePredictions = inputFilePredictions;
+            m_outputFolderPredictions = outputFolderPredictions;
+            m_predictionFailures = predictionFailures;
+        }
+
+        public void AddInputFile(string path, string projectDirectory, string predictorName)
+        {
+            if (!TryValidatePrediction(path, projectDirectory, predictorName, out string absolutePath))

```

Comment:
>TryValidatePrediction [](start = 17, length = 21)

Malformed paths were caught here before and reported as part of a failure, but now we are eating up the error. Any particular reason for changing this behavior?

Reply:
It's still being reported (see line 96). The validation we're losing is the one that checked that the path was absolute. This now makes the path absolute though if it's not already so that validation is no longer required.

---
In reply to: [294435653](https://github.com/microsoft/BuildXL/pull/480#discussion_r294435653) [](ancestors = 294435653)

Reply:
Ah, yes, you are right, ignore my comment

---
In reply to: [294438461](https://github.com/microsoft/BuildXL/pull/480#discussion_r294438461) [](ancestors = 294438461,294435653)